File: 0011-container-with-most-water/0011-container-with-most-water.swift
class Solution {
    func maxArea(_ height: [Int]) -> Int {
        var start = 0
        var end = height.count - 1
        var result = 0
    
        while start != end {
            let width = end - start
            let height1 = min(height[start], height[end])
            let water = width * height1

            if water > result {
                result = water
            }

            if height[start] > height[end] {
                end -= 1
            }else {
                start += 1
            }
            
        }
        return result
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        p1 = 0
        p2 = len(height) - 1
        record = 0

        while p1 < p2:
            h = min(height[p1], height[p2])
            w = (p2 - p1) * h
            record = max(record, w)
            print(p1, p2)
            if height[p1] >= height[p2]:
                p2 -= 1
            else:
                p1 += 1
        
        return record
